asv_bench/benchmarks/groupby.py

```diff
@@ -760,6 +760,8 @@ def setup(self, dtype, method):
         )
 
     def time_str_func(self, dtype, method):
+        if dtype == "string[python]" and method == "sum":
+            raise NotImplementedError
```

Comment:
Ah, so this is actually a change in behaviour also for the existing nullable string dtype:

```
In [11]: pd.Series(["a", "b"], dtype="string[python]").sum()
...
TypeError: Cannot perform reduction 'sum' with string dtype

In [12]: pd.Series(["a", "b"], dtype="string[python]").groupby([0, 0]).sum()
Out[12]: 
0    ab
dtype: string
```

But so it was also already inconsistent between plain sum and groupby sum.